tests/SpecTests/ClientSideEncryption/Prose22_RangeExplicitEncryptionTest.php

```diff
@@ -0,0 +1,448 @@
+    private ?Collection $collection = null;
+    private ?Client $encryptedClient = null;
+    private $key1Id;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        if ($this->isStandalone()) {
+            $this->markTestSkipped('Range explicit encryption tests require replica sets');
+        }
+
+        $this->skipIfServerVersion('<', '8.0.0', 'Range explicit encryption tests require MongoDB 8.0 or later');
+
+        $client = static::createTestClient();
+
+        $key1Document = $this->decodeJson(file_get_contents(__DIR__ . '/../client-side-encryption/etc/data/keys/key1-document.json'));
+        $this->key1Id = $key1Document->_id;
+
+        // Drop the key vault collection and insert key1Document with a majority write concern
+        self::insertKeyVaultData($client, [$key1Document]);
+
+        $this->clientEncryption = $client->createClientEncryption([
+            'keyVaultNamespace' => 'keyvault.datakeys',
+            'kmsProviders' => ['local' => ['key' => new Binary(base64_decode(self::LOCAL_MASTERKEY))]],
+        ]);
+
+        $autoEncryptionOpts = [
+            'keyVaultNamespace' => 'keyvault.datakeys',
+            'kmsProviders' => ['local' => ['key' => new Binary(base64_decode(self::LOCAL_MASTERKEY))]],
+            'bypassQueryAnalysis' => true,
+        ];
+
+        $this->encryptedClient = self::createTestClient(null, [], [
+            'autoEncryption' => $autoEncryptionOpts,
+            /* libmongocrypt caches results from listCollections. Use a new
+             * client in each test to ensure its encryptedFields is applied. */
+            'disableClientPersistence' => true,
+        ]);
+    }
+
+    public function setUpWithTypeAndRangeOpts(string $type, array $rangeOpts): void
+    {
+        if ($type === 'DecimalNoPrecision' || $type === 'DecimalPrecision') {
+            $this->markTestSkipped('Bundled libmongocrypt does not support Decimal128 (PHPC-2207)');
+        }
+
+        /* Read the encryptedFields file directly into BSON to preserve typing
+         * for 64-bit integers. This means that DropEncryptedCollection and
+         * CreateEncryptedCollection will be unable to inspect the option for
+         * metadata collection names, but that's not necessary for the test. */
+        $encryptedFields = Document::fromJSON(file_get_contents(__DIR__ . '/../client-side-encryption/etc/data/range-encryptedFields-' . $type . '.json'));
+
+        $database = $this->encryptedClient->selectDatabase($this->getDatabaseName());
+        $database->dropCollection('explicit_encryption', ['encryptedFields' => $encryptedFields]);
+        $database->createCollection('explicit_encryption', ['encryptedFields' => $encryptedFields]);
+        $this->collection = $database->selectCollection('explicit_encryption');
+
+        $encryptOpts = [
+            'keyId' => $this->key1Id,
+            'algorithm' => ClientEncryption::ALGORITHM_RANGE,
```

Comment:
This constant is unknown by PHPStorm stubs.

Reply:
These were introduced in 1.x by https://github.com/mongodb/mongo-php-driver/commit/4956b57028c3627b201b8e23720aa70206c13e13 and have yet to be released. I'm not sure what the process is for communicating such API changes downstream, but I don't recall us tracking that in JIRA for previous releases. It might have been something @alcaeus handled on the side.

Reply:
Yes, something we maintain: https://github.com/JetBrains/phpstorm-stubs/pulls?q=is%3Apr+mongodb